File: app/src/main/java/com/hlibrary/util/file/StreamTool.java
package com.hlibrary.util.file;

import android.support.annotation.NonNull;

import java.io.ByteArrayOutputStream;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

public class StreamTool {

    public static byte[] readStream(@NonNull String filename) throws IOException {
        FileInputStream in = new FileInputStream(filename);
        return readStream(in);
    }

    public static byte[] readStream(@NonNull InputStream in) throws IOException {
        byte[] data = new byte[1024];
        ByteArrayOutputStream bos = new ByteArrayOutputStream();
        int len;
        while ((len = in.read(data)) != -1) {
            bos.write(data, 0, len);
        }
        return bos.toByteArray();
    }

}

File: app/src/main/java/com/hlibrary/util/file/StreamTool.kt
package com.hlibrary.util.file

import java.io.ByteArrayOutputStream
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream

/**
 * @author linwh
 * @date 2015/01/23
 */
object StreamTool {

    @Throws(IOException::class)
    fun readStream(filename: String): ByteArray {
        val `in` = FileInputStream(filename)
        return readStream(`in`)
    }

    @Throws(IOException::class)
    fun readStream(inputStream: InputStream): ByteArray {
        val data = ByteArray(1024)
        val bos = ByteArrayOutputStream()
        var len: Int = inputStream.read(data)
        while (len != -1) {
            bos.write(data, 0, len)
            len = inputStream.read(data)
        }
        return bos.toByteArray()
    }

}
